src/opsinputs/opsinputs_cxwriter_mod.F90

```diff
@@ -1087,174 +1091,179 @@ subroutine opsinputs_cxwriter_populatecx(self, ReportFlags, Cx)
 
     ! Upper-air variables
     case (StashItem_theta) ! IndexCxtheta
-      call opsinputs_fill_fillreal2dfromgeoval( &
+      call opsinputs_fill_fillreal2dfromgeovalorhofx( &
         Cx % Header % theta, "theta", Cx % theta, &
         self % GeoVals, self % GeoVaLsAreTopToBottom, opsinputs_cxfields_theta, &
-        self % JediToOpsLayoutMapping, self % hofx, self % varnames)
+        self % JediToOpsLayoutMapping, self % hofx, self % varnames, &
+        "potentialTemperature", self % WriteHofXIntoCX)
```

Comment:
Note the extra options provided to theta, T, u, v and RH. These options are not needed otherwise.